src/classes/USER_UserService.cls

```diff
@@ -0,0 +1,271 @@
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+* @author Salesforce.org
+* @date 2017
+* @group User
+* @description Service Class to handle actions on the User object.
+*/
+public with sharing class USER_UserService {
+
+	//================================
+	// Constants
+	//================================
+	public static final String SYSTEM_ADMINISTRATOR = 'System Administrator';
+	public static final String OBJECT_ID_PREFIX_USER = '005';
+	public static final String ERROR_NOTIFICATION_RECIPIENT_ALL_SYS_ADMINS = 'All Sys Admins';
+	public static final String JOB_TYPE_SCHEDULED_APEX = '7';
+	public static final String ERROR_TYPE_SCHEDULED_APEX = 'Scheduled Apex Error';
+	public static final String OBJECT_TYPE_USER = 'User';
+	public static final String ERROR_MESSAGE_SCHEDULED_APEX = 'There are scheduled jobs owned by an inactive or frozen user. Review your scheduled jobs and ensure the owner of each one is an active user.';
+	public static final String CONTEXT_TYPE_USER = 'User';
+	
+
+	/*********************************************************************************************************
+    * @description Determines if each User in the List is a Sys Admin based on ProfileId.
+    * @param updatedUsers The users that were modified.
+    * @return A Map containing the UserId and Boolean value based on the User's Profile.
+    **********************************************************************************************************/
+	public static Map<Id, Boolean> isSysAdmin(List<User> updatedUsers) {
+		Map<Id, Boolean> userIdToIsSysAdmin = new Map<Id, Boolean>();
+		
+		// Retrieve the SysAdmin Profile Id
+		Id sysAdminId = retrieveSysAdminProfileId();
+
+		// Determine if the User is a SysAdmin
+		if(null != updatedUsers && !updatedUsers.isEmpty()) {
+			for(User currentUser : updatedUsers) {
+				if(sysAdminId == currentUser.ProfileId) {
+					userIdToIsSysAdmin.put(currentUser.Id, true);
+				} else {
+					userIdToIsSysAdmin.put(currentUser.Id, false);
+				}
+			}
+		}
+
+		return userIdToIsSysAdmin;
+	}
+
+
+	/*********************************************************************************************************
+    * @description Determines if each User in the List is active based on IsActive field.
+    * @param updatedUsers The users that were modified.
+    * @return A Map containing the UserId and Boolean value based on the User's IsActive status.
+    **********************************************************************************************************/
+	public static Map<Id, Boolean> isActiveUser(List<User> updatedUsers) {
+		Map<Id, Boolean> userIdToIsActive = new Map<Id, Boolean>();
+		
+		// Determine if the User isActive
+		if(null != updatedUsers && !updatedUsers.isEmpty()) {
+			for(User currentUser : updatedUsers) {
+				if(currentUser.IsActive) {
```

Comment:
i'd just go with userIdToIsActive.put(currentUser.Id, currentUser.IsActive);

Reply:
Implemented this code change. Thanks!